database/multistmt/parse.go

```diff
@@ -41,33 +43,53 @@ func Parse(reader io.Reader, delimiter []byte, maxMigrationSize int, h Handler)
 	// 3. could be converted to work with logger, for now fmt is still used
 	// 4. doesn't support /* */ c-style comments (future)
 	// 5. doesn't support nested comments (future)
+	// 6. now supports plpgsql trigger bodies
 	var err error = nil
 	// buf is the bytes read from input reader
 	buf := make([]byte, 1024)
 	// true when we're ignoring input(during comments)
 	discard := false
+	// fnbody is true when a function body delimiters $$ are encountered
+	fnbody := false
 	// accumulate statements intermediate buffer, this buffer will be incomplete
 	// until end-of-statement char ';'
-	accum := make([]byte, 0, 1024)
+	accum := make([]byte, 0, 2048)
 	// completed statements, contents of accum will be dumped in here
 	stmts := make([][]byte, 0, 1000)
+
 	for err == nil {
 		n, err := reader.Read(buf)
 		if n > 0 {
 			for i := range buf[:n] {
-				// when first two chars are comment indicators.
-				switch {
-				// ignore all lines that start with --
-				case len(buf) > 1 && i+1 < len(buf) && buf[i] == '-' && buf[i+1] == '-':
-					discard = true
-				// ignore any lines that start with // (this also covers ///)
-				case len(buf) > 1 && i+1 < len(buf) && buf[i] == '/' && buf[i+1] == '/':
-					discard = true
+				if !fnbody {
+					// when first two chars are comment indicators.
+					switch {
```

Comment:
What happens if there is a comment inside a fnbody?

Reply:
comments in fnbody are preserved and passed to the function(they will be part of the source) we want to keep those in source